app/models/group.rb

```diff
@@ -0,0 +1,7 @@
+class Group < ActiveRecord::Base
+  has_many :user_group_memberships
+  has_many :users, through: :user_group_memberships
+  belongs_to :moderator, class_name: :User, foreign_key: :moderator_id
```

Comment:
I think that's reasonable. The relationship they share is the one of a moderator. I think this route gives you greater flexibility for sure. :+1: